lib/chef/digester.rb

```diff
@@ -38,7 +38,11 @@ def checksum_for_file(file)
     end
 
     def generate_checksum(file)
-      checksum_file(file, OpenSSL::Digest::SHA256.new)
+      if file.is_a?(StringIO)
```

Comment:
Would it make sense to check for `respond_to?(:read)` ? I don't know why you'd need it, but it would allow you to `File.open(thing, opts) { |f| generate_checksum(f) }` (maybe that would be helpful if dealing with a gzipped file?).